PrestaShop/PrestaShop
# Migrate Cart Rules Create action

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Migrates form for existing AddCartRuleCommand (which was not yet used, but existed in develop branch).
| Type?             | refacto
| Category?         | BO
| BC breaks?        | see bellow
| Deprecations?     | no
| How to test?      | see bellow
| Fixed ticket?     | part of #10547
| Related PRs       | If theme, autoupgrade or other module change is needed to make this change work, provide a link to related PRs here.
| Sponsor company   | Your company or customer's name goes here (if applicable).

**BC BREAKS** (AddCartRuleCommand and CartRuleActionBuilder was not yet used anywhere except tests and the new page cart rule page wasn't yet exposed for usage in UX. Altough there are theoretical chances that somebody used it in module, because it wasn't marked as internal or so.)
Various property & methods naming changes, argument type and order changes in `PrestaShop\PrestaShop\Core\Domain\CartRule\Command\AddCartRuleCommand` and `PrestaShop\PrestaShop\Core\Domain\CartRule\Command\CartRuleActionBuilder` they might evolve even more during following PR's until the page is fully migrated, so I don't think its worth getting into details.

**HOW TO TEST**
The cart rule form is not finished, so no need to test it, but some product edition form parts were impacted so need to test them - in product page v2. Basically need to check that all fields that uses entitySearchComponent works as expected so there are no regressions:
 1. Description tab - related product search,
 2. "Pack" tab search for pack products when product type is pack,
 3. SEO tab redirect target product search when redirect to product option is selected.
 4. Details tab search files
 5. Pricing tab, when creating specific price - customer search

![Screenshot from 2023-03-31 18-18-24](https://user-images.githubusercontent.com/31609858/229161675-9493f6e9-7a24-4125-81cc-ae208c22b71c.png)
![Screenshot from 2023-03-31 18-18-27](https://user-images.githubusercontent.com/31609858/229161698-9af00b2d-b594-4556-8a2c-ad7b9e444d1b.png)
![Screenshot from 2023-03-31 18-18-44](https://user-images.githubusercontent.com/31609858/229161760-413dc195-a546-409b-80c5-90a897ff2eec.png)
